implementations/rust/ockam/ockam_identity/src/channel/secure_channel_worker.rs

```diff
@@ -150,7 +150,7 @@ impl<I: IdentityTrait, T: TrustPolicy> SecureChannelWorker<I, T> {
     pub(crate) async fn create_responder(
         ctx: &Context,
         identity: I,
-        trust_policy: T,
+        trust_policy: Arc<dyn TrustPolicy>,
```

Comment:
This is different approach comparing to what we use everywhere else, why have you made that decision?

Reply:
Imagine a user needs to support multiple kinds of TrustPolicy at the same time. For example, if the CLI had an arg that was `--trust-everyone` that overrode the file choice.

This way, they can just do something 
```rs
fn get_trust_policy() -> Arc<dyn TrustPolicy> {
    // this wouldn't be how we actually do but you get the idea
    if given_arg("--trust-everyone") {
        Arc::new(TrustEveryonePolicy)
    } else {
        Arc::new(TrustMultiIdentifiersPolicy::new(get_identifiers()))
    }
}
```

The other option would be to force them to write an enum like
```rs
pub enum MyTrustPolicy {
    Everyone(TrustEveryonePolicy),
    List(TrustMultiIdentifiersPolicy),
    // ...
}
```

But this requires much more boilerplate, and is more tedious.

Reply:
I think this case is cut and dry enough to keep this way, TBH. I don't think we need to make a decision like this universally.